bundles/org.openhab.binding.linky/src/main/java/org/openhab/binding/linky/internal/handler/LinkyHandler.java

```diff
@@ -273,22 +275,15 @@ private synchronized void updateMonthlyData() {
      */
     private synchronized void updateYearlyData() {
         if (isLinked(LAST_YEAR) || isLinked(THIS_YEAR)) {
-            cachedYearlyData.getValue().ifPresentOrElse(values -> {
-                Aggregate years = values.aggregats.years;
-                updateKwhChannel(LAST_YEAR, years.datas.get(0));
-                if (years.datas.size() > 1) {
-                    updateKwhChannel(THIS_YEAR, years.datas.get(1));
-                } else {
-                    updateKwhChannel(THIS_YEAR, 0.0);
-                }
-            }, () -> {
-                if (ZonedDateTime.now().getDayOfYear() == 1) {
-                    updateKwhChannel(THIS_YEAR, 0.0);
-                    updateKwhChannel(LAST_YEAR, Double.NaN);
-                } else {
-                    updateKwhChannel(THIS_YEAR, Double.NaN);
-                }
-            });
+            cachedYearlyData.getValue().ifPresentOrElse(
+                    values -> setCurrentAndPrevious(values.aggregats.years, THIS_MONTH, LAST_MONTH), () -> {
```

Comment:
Wrong channels: use THIS_YEAR and THIS_YEAR.

Reply:
Corrected.